tests/conftest.py

```diff
@@ -58,19 +59,38 @@ def test_with_build_secrets(...)
     return features
 
 
+class FakeConfigModel(craft_application.ConfigModel):
```

Comment:
Maybe it should also include an `Enum` that stores `int` / `str` example, as you're treating enums differently during type casting ?
I know that it should be covered by EmitterMode, but explicit test would be nice

Reply:
Added